benchmark/cublas/blas2/gbmv.cpp

```diff
@@ -0,0 +1,202 @@
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gbmv.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t, int m, int n, int kl, int ku) {
+  std::ostringstream str{};
+  str << "BM_Gbmv<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << t << "/" << m << "/" << n << "/" << kl << "/" << ku;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSgbmv(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDgbmv(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr, int ti,
+         index_t m, index_t n, index_t kl, index_t ku, scalar_t alpha,
+         scalar_t beta, bool* success) {
+  // Standard test setup.
+  std::string ts = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(ti));
+  const char* t_str = ts.c_str();
+
+  index_t xlen = t_str[0] == 'n' ? n : m;
+  index_t ylen = t_str[0] == 'n' ? m : n;
+
+  index_t lda = (kl + ku + 1);
+  index_t incX = 1;
+  index_t incY = 1;
+
+  // The counters are double. We convert m and n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double m_d = static_cast<double>(m);
+  double n_d = static_cast<double>(n);
+  double kl_d = static_cast<double>(kl);
+  double ku_d = static_cast<double>(ku);
+
+  state.counters["m"] = m_d;
+  state.counters["n"] = n_d;
+  state.counters["kl"] = kl_d;
+  state.counters["ku"] = ku_d;
+
+  // Compute the number of A non-zero elements.
+  const double A_validVal =
+      (t_str[0] == 'n' ? n_d : m_d) * (kl_d + ku_d + 1.0) -
+      0.5 * (kl_d * (kl_d + 1.0)) - 0.5 * (ku_d * (ku_d + 1.0));
+
+  {
+    double nflops_AtimesX = 2.0 * A_validVal;
+    double nflops_timesAlpha = ylen;
+    double nflops_addBetaY = (beta != scalar_t{0}) ? 2 * ylen : 0;
+    state.counters["n_fl_ops"] =
+        nflops_AtimesX + nflops_timesAlpha + nflops_addBetaY;
+  }
+  {
+    double mem_readA = A_validVal;
+    double mem_readX = xlen;
+    double mem_writeY = ylen;
+    double mem_readY = (beta != scalar_t{0}) ? ylen : 0;
+    state.counters["bytes_processed"] =
+        (mem_readA + mem_readX + mem_writeY + mem_readY) * sizeof(scalar_t);
+  }
+
```

Comment:
Can you also add this line (and line 126) in these Level 2 benchmarks? This basically reports the `items_processed_per_second` which we use to check the performance for the operator on the hardware.
https://github.com/codeplaysoftware/sycl-blas/blob/adc04a93803199669afe29dae1bdda010da9d00e/benchmark/syclblas/blas3/gemm.cpp#L133